src/Microsoft.OData.Core/ODataBatchWriter.cs

```diff
@@ -447,26 +363,9 @@ private void VerifyCanWriteEndBatch(bool synchronousCall)
         }
 
         /// <summary>
-        /// Ends a batch - implementation of the actual functionality.
+        /// Ends a batch.
         /// </summary>
-        private void WriteEndBatchImplementation()
-        {
-            Debug.Assert(
-                this.batchStartBoundaryWritten || this.CurrentOperationMessage == null,
-                "If not batch boundary was written we must not have an active message.");
-
-            // write pending message data (headers, response line) for a previously unclosed message/request
-            this.WritePendingMessageData(true);
-
-            this.SetState(BatchWriterState.BatchCompleted);
-
-            // write the end boundary for the batch
-            ODataBatchWriterUtils.WriteEndBoundary(this.rawOutputContext.TextWriter, this.batchBoundary, !this.batchStartBoundaryWritten);
-
-            // For compatibility with WCF DS we write a newline after the end batch boundary.
-            // Technically it's not needed, but it doesn't violate anything either.
-            this.rawOutputContext.TextWriter.WriteLine();
-        }
+        internal abstract void WriteEndBatchImplementation();

```

Comment:
Implementation methods are not generally in base classes, they are called from the derived classe's override mehtod. -- is there a reason to have this here?